test-data/unit/check-selftype.test

```diff
@@ -802,6 +802,37 @@ class Bad(metaclass=Meta):
 Good.do_x()
 Bad.do_x()  # E: Invalid self argument "Type[Bad]" to attribute function "do_x" with type "Callable[[Type[T]], T]"
 
+[case testSelfTypeProtocolClassmethodMatch]
+from typing import Type, TypeVar, Protocol
+
+T = TypeVar('T')
+
+class HasDoX(Protocol):
+    @classmethod
```

Comment:
What about `@staticmethod`? Does it fall into the same problem?

Reply:
staticmethod doesn't have this problem, because there's no self/cls to bind